jsapp/js/components/map.es6

```diff
@@ -322,7 +361,35 @@ export class FormMap extends React.Component {
     this.setState({
       showExpandedLegend: !this.state.showExpandedLegend,
     });
+  }
 
+  filterByMarker(evt) {
+    let markers = this.state.markers,
+        id = evt.target.getAttribute('data-id'),
+        filteredByMarker = this.state.filteredByMarker;
+
+    if (!filteredByMarker)
+      filteredByMarker = [id];
+    else if (!filteredByMarker.includes(id))
+      filteredByMarker.push(id);
+    else
+      filteredByMarker = filteredByMarker.filter(l => l !== id);
+
+    this.setState({filteredByMarker: filteredByMarker});
+    markers.eachLayer( function(layer) {
+      if (!filteredByMarker.includes(layer.options.typeId.toString()))
+        layer._icon.classList.add("unselected");
+      else
+        layer._icon.classList.remove("unselected");
+    });
+  }
+
+  resetFilterByMarker() {
+    let markers = this.state.markers;
+    this.setState({filteredByMarker: false});
+    markers.eachLayer( function(layer) {
+      layer._icon.classList.remove("unselected");
```

Comment:
"unselected" is used few times, I think it would be safer as a declared variable?